packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -428,9 +493,23 @@ class AuroraPostgresClusterEngine extends ClusterEngineBase {
           majorVersion: version.auroraPostgresMajorVersion,
         }
         : undefined,
+      features: version?._features,
     });
   }
 
+  public bindToCluster(scope: core.Construct, options: ClusterEngineBindOptions): ClusterEngineConfig {
+    const config = super.bindToCluster(scope, options);
+    if (options.s3ImportRole && !(config.features?.s3Import)) {
+      throw new Error (`s3Import is not supported for engine: ${this.engineVersion?.fullVersion}. Use an engine version that supports the s3Import feature.`);
+    }
+    if (options.s3ExportRole && !(config.features?.s3Export)) {
```

Comment:
These `if`s here mean that the unversioned Postgres engine doesn't support S3 import/export at all. Is that true? Have you verified it?

Reply:
you're right that this code precludes supporting these features for the unversioned Postgres version.
The only reason I added these conditions are because it's a latent failure and it felt like adding validation would help users in getting ahead of it.

The reason I left out the unversioned feature support is because we've already marked them deprecated and the feature is not supported today. I didn't think it was a good idea to try and support it and keep that usage going and was hoping to nudge users towards the guidance we're providing in our doc string.

What do you think? is this validation worth it? should we drop it altogether?
This may also affect the default message for the error (should probably drop the `?? <unversioned>`).

Reply:
My opinion is the following: if the RDS service supports it, we should support it as well.

Note that we might also have to "un-deprecate" these fields at some point before CDK v2.0 hits - see the discussion [here](https://github.com/aws/aws-cdk-rfcs/pull/241#discussion_r480448417), which makes fully supporting them even more relevant.

Reply:
gotcha, I didn't realize we weren't fully convinced that we wanted to keep it marked as `@deprecated`
added support for this case and skipped validation if the featurename is supported.

When unversioned is specified, both features are marked as supported. This is true today, but may not be if the version is changed out from underneath. I'm not sure the constant can support it both ways

Reply:
I would mark is as supported for both, always (otherwise, this feature has no chance of working with the unversioned engine at all).

Worst comes to worst, it will fail at deploy time, which is fine by me.

Reply:
Actually, here's a thought.

Today, you always return the features names in the `features` property of `ClusterEngineConfig`. I wonder whether a better strategy would be to only return them if `s3ImportRole` / `s3ExportRole` was passed in `ClusterEngineBindOptions`...?

Yes, I understand the code in Cluster today uses the feature names only if `s3ImportRole` / `s3ExportRole`  are != `undefined`... but I wonder whether that will e obvious when we implement other feature names, and whether it's actually better to not return them if they were not requested...

Thoughts on this change?